src/m365/spo/commands/list/list-remove.spec.ts

```diff
@@ -138,15 +132,41 @@ describe(commands.LIST_REMOVE, () => {
     await command.action(logger, { options: { id: 'b2307a39-e878-458b-bc90-03bc578531d6', webUrl: 'https://contoso.sharepoint.com', force: true } });
     let correctRequestIssued = false;
     requests.forEach(r => {
-      if (r.url.indexOf(`/_api/web/lists(guid'`) > -1 &&
-        r.headers.accept &&
-        r.headers.accept.indexOf('application/json') === 0) {
+      if (r.url === `${webUrl}/_api/web/lists(guid'${listId}')`) {
         correctRequestIssued = true;
       }
     });
     assert(correctRequestIssued);
   });
 
+  it('uses correct API url when recycle option is passed', async () => {
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      requests.push(opts);
+      if (opts.url === `${webUrl}/_api/web/lists/GetByTitle('${listTitle}')/recycle`) {
+        return 'Correct URL';
+      }
+
+      throw 'Invalid request';
+    });
+    sinonUtil.restore(cli.promptForConfirmation);
+    sinon.stub(cli, 'promptForConfirmation').resolves(true);
+    await command.action(logger, {
+      options: {
+        title: 'Documents',
+        recycle: true,
+        webUrl: 'https://contoso.sharepoint.com'
+      }
+    });
+    let correctRequestIssued = false;
+    requests.forEach(r => {
+      if (r.url === `${webUrl}/_api/web/lists/GetByTitle('${listTitle}')/recycle`) {
+        correctRequestIssued = true;
+      }
+    });
```

Comment:
Instead of doing this, let's use `assert(postStub.calledOnce)`. This ensures that the request was called once and didn't throw an error.